Edits for BOLT-07

shannona opened the pull request:
A variety of edits for clarity and our styles sheets for BOLT-7. The one other issue for BOLT-7 has been filed as issue #302.

rustyrussell commented on 07-routing-gossip.md:
```diff
@@ -1,19 +1,25 @@
-but multiple `node_announcement` messages are possible (to update node
-information), and at least two `channel_update` messages are expected.
+To support node discovery, peers exchange `node_announcement`
+messages, which supply additional information about nodes There can be
```
Missing period between "about nodes" and "There can be"

shannona replied:
Fixed.

rustyrussell commented on 07-routing-gossip.md:
```diff
@@ -28,49 +34,50 @@ The willingness of the initiating node to announce the channel is signaled durin
-index which pays to the channel.
+index that pays to the channel.
+
+If the `open_channel` message had the `announce_channel` bit set, then both nodes must send the `announcement_signatures` message, otherwise they MUST NOT.
```
must -> MUST.

shannona replied:
Fixed.

rustyrussell commented on 07-routing-gossip.md:
```diff
@@ -474,13 +479,13 @@ channel:
-1. D: 400 base + 4000 millionths
+2. B: 200 base + 2000 millionths
+3. C: 300 base + 3000 millionths
+4. D: 400 base + 4000 millionths
```
This is fine, but FYI, markdown turns these into an enumerated list when rendering anyway :)

toadlyBroodle commented on 07-routing-gossip.md:
```diff
@@ -28,49 +34,50 @@ The willingness of the initiating node to announce the channel is signaled durin
-If the `open_channel` message had the `announce_channel` bit set, then both nodes must send the `announcement_signatures` message, otherwise they MUST NOT.
-
-If sent, `announcement_signatures` messages MUST NOT be sent until `funding_locked` has been sent and the funding transaction has at least 6 confirmations.
+The `announcement_signatures` message is created by constructing a `channel_announcement` message corresponding to the newly established channel and signing it with the secrets matching an endpoint's `node_id` and `bitcoin_key`. The message is then sent using an `announcement_signatures`.
```
...message, corresponding...channel, and...

shannona replied:
Yep, clearer. Fixed.

toadlyBroodle commented on 07-routing-gossip.md:
```diff
@@ -28,49 +34,50 @@ The willingness of the initiating node to announce the channel is signaled durin
+
+If the `open_channel` message had the `announce_channel` bit set, then both nodes must send the `announcement_signatures` message, otherwise they MUST NOT.
+
+If sent, `announcement_signatures` messages MUST NOT be sent until `funding_locked` has been sent and the funding transaction has at least 6 confirmations.
```
six

shannona replied:
Fixed.

toadlyBroodle commented on 07-routing-gossip.md:
```diff
@@ -28,49 +34,50 @@ The willingness of the initiating node to announce the channel is signaled durin
+`node_2` requires:
 
-1. Prove that the funding transaction pays to `bitcoin_key_1` and
+1. Proving that the funding transaction pays to `bitcoin_key_1` and
```
proving...`bitcoin_key_1`  <--(no 'and')

shannona replied:
The whole line is actually: "1. Proving that the funding transaction pays to `bitcoin_key_1` and
   `bitcoin_key_2`."Unless there's a style I'm missing, kept.

toadlyBroodle commented on 07-routing-gossip.md:
```diff
@@ -28,49 +34,50 @@ The willingness of the initiating node to announce the channel is signaled durin
-2. Prove `node_1` owns `bitcoin_key_1`
-3. Prove `node_2` owns `bitcoin_key_2`
+2. Proving that `node_1` owns `bitcoin_key_1`
+3. Proving that `node_2` owns `bitcoin_key_2`
```
proving...

shannona replied:
I think I didn't have that style update when I did 07, so keep an eye out when you get there. In any case, I've fixed all three of these lines to be no initial cap and no end period,

shannona replied:
(Actually, I reskimmed, and this is the only list with fragments, so we should have this now.

toadlyBroodle commented on 07-routing-gossip.md:
```diff
@@ -28,49 +34,50 @@ The willingness of the initiating node to announce the channel is signaled durin
-We also need to prove that `node_1` and `node_2` both agree on this
-announcement message; that is done by having a signature from each
+It is also necessary to prove that `node_1` and `node_2` both agree on this
+announcement message; that is accomplished by having a signature from each
```
...message; this is...

shannona replied:
Done.

toadlyBroodle commented on 07-routing-gossip.md:
```diff
@@ -256,50 +263,51 @@ if `port` is zero.
 announces the fees and minimum expiry delta it requires to relay HTLCs
-through this channel.  It uses the 8-byte
-channel shortid which matches the `channel_announcement` and one bit
+through this channel. They use the 8-byte
```
...channel. Each uses the...

shannona replied:
Changed.

toadlyBroodle commented on 07-routing-gossip.md:
```diff
@@ -256,50 +263,51 @@ if `port` is zero.
-through this channel.  It uses the 8-byte
-channel shortid which matches the `channel_announcement` and one bit
+through this channel. They use the 8-byte
+channel shortid that matches the `channel_announcement` and one bit
```
... and 1 bit

shannona replied:
Fixed.

shannona commented:
Updated commit: (1) addresses all reviews from @rustyrussell and @toadlyBroodle ; (2) Correctly capitalizes a few sections that had been missed; and (3) incorporates table of contents courtesy of @bcongdon in #310
